iceoryx_binding_c/source/c_publisher.cpp

```diff
@@ -30,25 +31,63 @@ extern "C" {
 #include "iceoryx_binding_c/publisher.h"
 }
 
+constexpr uint64_t PUBLISHER_OPTIONS_INIT_CHECK_CONSTANT = 123454321;
+
+void iox_pub_options_init(iox_pub_options_t* options)
+{
+    if (options == nullptr)
+    {
+        LogWarn() << "publisher options initialization skipped - null pointer provided";
+        return;
+    }
+
+    PublisherOptions publisherOptions;
+    options->historyCapacity = publisherOptions.historyCapacity;
+    options->nodeName = nullptr;
+
+    options->initCheck = PUBLISHER_OPTIONS_INIT_CHECK_CONSTANT;
+}
+
+bool iox_pub_options_is_initialized(const iox_pub_options_t* const options)
+{
+    return options && options->initCheck == PUBLISHER_OPTIONS_INIT_CHECK_CONSTANT;
+}
+
 iox_pub_t iox_pub_init(iox_pub_storage_t* self,
                        const char* const service,
                        const char* const instance,
                        const char* const event,
-                       const uint64_t historyCapacity,
-                       const char* const nodeName)
+                       const iox_pub_options_t* const options)
 {
     new (self) cpp2c_Publisher();
     iox_pub_t me = reinterpret_cast<iox_pub_t>(self);
-    PublisherOptions options;
-    options.historyCapacity = historyCapacity;
-    options.nodeName = NodeName_t(TruncateToCapacity, nodeName);
+    PublisherOptions publisherOptions;
+
+    // use default options otherwise
+    if (options != nullptr)
+    {
+        if (!iox_pub_options_is_initialized(options))
+        {
+            std::cout << "******* terminate" << std::endl;
+            // note that they may have been initialized but the initCheck
+            // pattern overwritten afterwards, we cannot be sure but it is a misuse
+            LogError() << "publisher options may not have been initialized with iox_pub_init";
+            std::terminate();
```

Comment:
`LogFatal` should be used here and I guess also the error handler according to the error handling document

Reply:
Done. But Note that we have some more terminates in the C Binding and I am not sure whether we want the error handling rules also for Bindings, Gateways and so on but only the core. 
E.g. in utils we also have a different approach (since it is a generic library).